### ISSUE TITLE
🚀 Feature: Autofixer and/or suggestions in inferable cases

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [es-tooling/eslint-plugin-depend#40]
Manually replacing existing dependencies is a painful slog. It'd be lovely to have automated assistance from the lint rule.

My initial instinct is that there are three kinds of cases:

* [Fixer-level](https://eslint.org/docs/latest/use/core-concepts/glossary#fix): APIs that are directly equivalent and can be switched to without change in behavior[^1]
* [Suggestion-level](https://eslint.org/docs/latest/use/core-concepts/glossary#suggestion): APIs that are generally the same and _probably_ safe to switch to, but need manual review
* Very different ones that are either too functionally different or too difficult to automate

Feature request: can this be added to the rule?

[^1]: Or are at least _intended_ to be exactly the same. Even "safe" fixers sometimes can technically break teeny edge cases.

**Comment by JoshuaKGoldberg** [comment]
Yeah codemods on their own are nice, but IME most developers & projects don't go the extra mile to use them. Lint rules are _right there_ in the editor. 

From another perspective: ESLint is actually a pretty good codemod platform (fixes & suggestions, inline disable comments, configuration overrides with glob matchers, ...). And it keeps things fixed over time by blocking introduced violations. I find it hard to advocate for any other codemod system 😄

**Comment by 43081j** [comment]
that's true

it would be ideal I think if we could share the codemod logic between the linter and a future codemod cli

especially since the codemods repo is already consumed by some things

I'll have a ponder on it
